widget/entry.go

```diff
@@ -893,11 +894,13 @@ func (e *Entry) TypedRune(r rune) {
 	})
 	e.propertyLock.Unlock()
 
-	e.Validate()
+	if err := e.Validate(); err == nil || errors.Is(err, e.validationError) {
+		// Under this situation, the validate function won't call refresh, so do it here
```

Comment:
You are hard-coding an assumption about the method you are calling. This is bound to be problematic when someone changes the method's behaviour and forgets to update all occurrences of this assumption. It will basically introduce code-rot in the long run.

Reply:
Yes. So introduce the validate() without refresh might be a better option.